lib/OpenQA/Worker/Jobs.pm

```diff
@@ -825,14 +829,21 @@ sub read_result_file($$) {
 }
 
 sub backend_running {
-    return $worker;
+    return $worker->{child}->is_running;
 }
 
 sub check_backend {
     log_debug("checking backend state");
-    my $res = engine_check;
-    if ($res && $res ne 'ok') {
-        stop_job($res);
+    unless ($worker->{child}->is_running()) {
+        if ($worker->{child}->errored) {
+            stop_job('died');
+        }
+        else {
+            stop_job('done');
+        }
```

Comment:
I always have the personal preference of a more concise style: `stop_job($worker->{child}->errored ? 'died' : 'done')` but as I said ... personal preference ;)